src/BlackHole.php

```diff
@@ -42,24 +42,20 @@ final class BlackHole implements
 {
     /**
      * Capabilities of this adapter
-     *
-     * @var null|Capabilities
      */
-    protected $capabilities;
+    private ?Capabilities $capabilities = null;
 
     /**
      * Marker to change capabilities
-     *
-     * @var null|object
      */
-    protected $capabilityMarker;
+    private ?stdClass $capabilityMarker = null;
 
     /**
      * options
      *
      * @var null|AdapterOptions
      */
-    protected $options;
+    private $options;
```

Comment:
Worth using a native type?

Reply:
Ok, updated with native types https://github.com/laminas/laminas-cache-storage-adapter-blackhole/pull/56/commits/876192e83e181d60f19850c27a451fea224e46a3

Reply:
it seems cause error in pslam as filled with mixed in `__construct()`

```
E░░░░░░░░░░
Error: src/BlackHole.php:83:33: PossiblyInvalidMethodCall: Cannot call method on possible non-empty-array<array-key, mixed> variable $this->options (see https://psalm.dev/113)
Error: src/BlackHole.php:83:33: PossiblyUndefinedMethod: Method Traversable::setAdapter does not exist (see https://psalm.dev/108)
Error: src/BlackHole.php:95:16: InvalidReturnType: The declared return type 'Laminas\Cache\Storage\Adapter\AdapterOptions' for Laminas\Cache\Storage\Adapter\BlackHole::getOptions is incorrect, got 'Laminas\Cache\Storage\Adapter\AdapterOptions|Laminas\Cache\Storage\Adapter\AdapterOptions&Traversable|Traversable|non-empty-array<array-key, mixed>' (see https://psalm.dev/011)
Error: src/BlackHole.php:104:16: InvalidReturnStatement: The inferred type 'Laminas\Cache\Storage\Adapter\AdapterOptions|Laminas\Cache\Storage\Adapter\AdapterOptions&Traversable|Traversable|non-empty-array<array-key, mixed>' does not match the declared return type 'Laminas\Cache\Storage\Adapter\AdapterOptions' for Laminas\Cache\Storage\Adapter\BlackHole::getOptions (see https://psalm.dev/128)

```

Rolling back https://github.com/laminas/laminas-cache-storage-adapter-blackhole/pull/56/commits/9f8f1d4edcd6c06493f201b5790aa6f2352d535a

Reply:
Fair enough: let's keep it for now :)